src/main/java/org/mule/service/soap/introspection/WsdlSchemasCollector.java

```diff
@@ -32,43 +35,71 @@
 @SuppressWarnings("unchecked")
 final class WsdlSchemasCollector {
 
-  SchemaCollector collect(Definition wsdlDefinition) {
-    SchemaCollector collector = SchemaCollector.getInstance();
+  private final Map<String, Schema> schemas = newHashMap();
+  private final Definition definition;
 
-    collectTypes(wsdlDefinition.getTypes(), collector);
+  WsdlSchemasCollector(Definition definition) {
+    this.definition = definition;
+  }
 
-    wsdlDefinition.getImports().values().forEach(wsdlImport -> {
-      Definition definition = ((Import) wsdlImport).getDefinition();
-      collectTypes(definition.getTypes(), collector);
+  public SchemaCollector collect() {
+    SchemaCollector collector = SchemaCollector.getInstance();
```

Comment:
if the definition is static, then do this in a LazyValue, so that you only do this once